### ISSUE TITLE
Need help to update 0.6.1 => 1.0.0

### ISSUE THREAD

**Issue by AlexandreBourdeaudhui** [novnc/noVNC#1140]
Hello 👋

Until now, I was using `noVNC@^0.6.1` in my React project. Today, i want to update this package in version `1.0.0`. After this update, when i tried to run again my app, i got this message `Uncaught SyntaxError: Unexpected token *`.

I saw [this doc](https://github.com/marck1391/noVNC/blob/master/docs/LIBRARY.md#conversion-of-modules) and tried it, but i don't know what i need to do with this...

If i'm trying to copy/paste `lib` folder in the `vendor` folder... I got a `base64.js:3 Uncaught ReferenceError: exports is not defined` when i'm trying to run my project...

If i'm creating a `libs` folder in my project and copy/paste the `lib` folder, i got this : `error: Processing of app/src/libs/noVNC/inflator.js failed. Error: Could not load module '../lib/vendor/pako/lib/zlib/inflate.js' from 'app/src/libs/noVNC'. Make sure the file actually exists.`

**Comment by AlexandreBourdeaudhui** [comment]
At this moment, i had to fork this project, compile it in `commonjs` and import the fork in my project.

**Comment by CendioOssman** [comment]
Hmmm... It seems an unconverted tree was pushed to npm. I have no idea how this is supposed to work.

@DirectXMan12, could you have a look?

**Comment by AlexandreBourdeaudhui** [comment]
Hello @CendioOssman 👋 and thanks for your response !

In fact, i can't use the fork with the [`conversion modules`](https://github.com/marck1391/noVNC/blob/master/docs/LIBRARY.md#conversion-of-modules) option.

When i try to share my screen, i got an error `l.Warn is not a function` or `l.Error is not a function`.

However, if i'm dropping `console` option to the uglify module, that's fine. But i got a message in Chrome console `Server native red-shift is not 16` and `Server native blue-shift is not 0`.

**Comment by CendioOssman** [comment]
Yeah, we've had other reports that uglify doesn't strip out the console properly. I'm not sure if anyone reported it to them though...

Those other messages are just warnings and can be ignored. Are you still seeing some functional issues?

**Comment by DirectXMan12** [comment]
the NPM module is *supposed* to have both a commonjs form and the raw source, since people using ES6 modules natively want to just use the raw source.  If that's not the case, we have issues.

**Comment by CendioOssman** [comment]
Then we do indeed have issues. This is all you get from npm:

```
$ ls node_modules/@novnc/novnc
core  docs  LICENSE.txt  package.json  README.md  vendor
```

**Comment by AlexandreBourdeaudhui** [comment]
> Are you still seeing some functional issues?

I don't see more functional issues ! Just what i reported in my last message

**Comment by CendioOssman** [comment]
Great, thanks. So then we can focus on just the publishing issue.

**Comment by DirectXMan12** [comment]
@CendioOssman oof, ok.  I'll try to take a look at what's going on next week (I'll be out of the office tomorrow ;-) )

**Comment by AlexandreBourdeaudhui** [comment]
Hello @DirectXMan12 👋 
Have you had time to take a look? 🙏

**Comment by DirectXMan12** [comment]
hey, sorry about that, I have not.  it's been an eventful past couple of months for me, and my time for noVNC has been significantly less than I would have liked

**Comment by CendioOssman** [comment]
Do you think you'll still manage to have a look at this? Or should someone else give it a go?

**Comment by DirectXMan12** [comment]
I'd prefer if someone else could give it a go (I'm swamped at work right now + unpacking boxes from moving), but let me know if you have questions, etc.  Basically, the NPM build process is supposed to put the commonjs code in one location, and the raw ES6 code in another, and mark the difference using attributes in package.json.

**Comment by CendioOssman** [comment]
I have no idea why this failed. It works just fine when I do it manually. And the job log on Travis shows basically no details at all.

Unfortunately the system isn't particularly helpful in doing test publishings either, so debugging is a bit of a pain... I'll see if I can get something working.

But I did see some issues with the fact that we rely on `.npmignore`. That tends to include a lot of gunk by accident. So I would suggest switching to using `"files"`:

```diff
diff --git a/package.json b/package.json
index 24cb763..ebd431b 100644
--- a/package.json
+++ b/package.json
@@ -2,10 +2,22 @@
   "name": "@novnc/novnc",
   "version": "1.0.0",
   "description": "An HTML5 VNC client",
+  "browser": "lib/rfb",
   "directories": {
+    "lib": "lib",
     "doc": "docs",
     "test": "tests"
   },
+  "files": [
+    "lib",
+    "AUTHORS",
+    "VERSION",
+    "docs/API.md",
+    "docs/LIBRARY.md",
+    "docs/LICENSE*",
+    "core",
+    "vendor/pako"
+  ],
   "scripts": {
     "lint": "eslint app core po tests utils",
     "test": "karma start karma.conf.js",
```

I also added `"browser"` as that seemed appropriate for our project.

Do these changes seem reasonable @DirectXMan12 and @kanaka?

**Comment by CendioOssman** [comment]
Alright, so the core issue has been identified. Travis has such an ancient npm that it does not support the `"prepare"` event. We'll have to use something older.

Before I start digging in to that though, is it a key feature that it is run on `npm install`? Or is it sufficient that it is used only for publishing? I'm not familiar with the use cases here. @DirectXMan12?

**Comment by DirectXMan12** [comment]
I think it's semi-expected to work on npm install to, but I'm a little rusty.  I don't recall exactly why we ran it on install too (may have been for the "install from git" usecase).

**Comment by CendioOssman** [comment]
This has been fixed on `master` now. Unfortunately npm doesn't allow us to fix already existing versions, so this won't be resolved until we publish 1.1.0. But we should probably have a look at that soon anyway.

**Comment by AlexandreBourdeaudhui** [comment]
Thanks you guys ! 👌 👌 👌
